kiliautoml/utils/cleanlab/train_cleanlab.py

```diff
@@ -161,138 +62,144 @@ def combine_folds(data_dir, model_dir, verbose=0, num_classes=10, nb_folds=4, se
     return destination
 
 
-def train_and_get_error_labels(
-    cv_n_folds,
-    data_dir,
-    epochs,
-    model_dir,
-    model_name,
-    verbose=0,
-    cv_seed=42,
-):
-    """
-    Main method that trains the model on the assets that are in data_dir, computes the
-    incorrect labels using Cleanlab and returns the IDs of the concerned assets.
-    """
-    # To set to False if the input size varies a lot and you see that the training takes
-    # too much time
-    cudnn.benchmark = True
+def single_true(iterable):
+    i = iter(iterable)
+    return any(i) and not any(i)
 
-    data_transforms = {
-        "train": transforms.Compose(
-            [
-                transforms.RandomResizedCrop(224),
-                transforms.RandomHorizontalFlip(),
-                transforms.ToTensor(),
-                transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
-            ]
-        ),
-        "val": transforms.Compose(
-            [
-                transforms.Resize(256),
-                transforms.CenterCrop(224),
-                transforms.ToTensor(),
-                transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
-            ]
-        ),
-    }
-
-    original_image_datasets = {
-        x: datasets.ImageFolder(data_dir, data_transforms[x]) for x in ["train", "val"]
-    }
 
-    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
-    class_names = original_image_datasets["train"].classes
-    labels = [label for img, label in datasets.ImageFolder(data_dir).imgs]
-    for cv_fold in range(cv_n_folds):
-        # Split train into train and holdout for particular cv_fold.
-        kf = StratifiedKFold(n_splits=cv_n_folds, shuffle=True, random_state=cv_seed)
-        cv_train_idx, cv_holdout_idx = list(kf.split(range(len(labels)), labels))[cv_fold]
-        # Separate datasets
-        np.random.seed(cv_seed)
-        train_idx, val_idx = train_test_split(cv_train_idx, test_size=0.2)
-        image_datasets = copy.deepcopy(original_image_datasets)
-        holdout_dataset = copy.deepcopy(image_datasets["val"])
-        holdout_dataset.imgs = [holdout_dataset.imgs[i] for i in cv_holdout_idx]
-        holdout_dataset.samples = holdout_dataset.imgs
-        image_datasets["train"].imgs = [image_datasets["train"].imgs[i] for i in train_idx]
-        image_datasets["train"].samples = image_datasets["train"].imgs
-        image_datasets["val"].imgs = [image_datasets["val"].imgs[i] for i in val_idx]
-        image_datasets["val"].samples = image_datasets["val"].imgs
+class ImageClassificationModel(BaseModel):
```

Comment:
to move to its own file in `kiliautoml.models`

Reply:
rename it to `PyTorchVisionImageClassificationModel`

Reply:
and I know it is not ideal, but put the arguments in the same order as the HuggingFace ones